src/placement/ring_map.c

```diff
@@ -1385,8 +1388,58 @@ ring_obj_find_rebuild(struct pl_map *map, struct daos_obj_md *md,
 			 * for rebuild, perhaps they should be unified.
 			 */
 			if (l_shard->po_shard != -1) {
+				struct pool_target	*target;
+				int			 leader;
+
 				D_ASSERT(f_shard->rfs_tgt_id != -1);
 				D_ASSERT(idx < array_size);
+
+				/* If the caller does not care about DTX related
+				 * things (myrank == -1), then fill it directly.
+				 */
+				if (myrank == -1)
+					goto fill;
+
+				leader = pl_select_leader(md->omd_id,
+					l_shard->po_shard, layout->ol_nr,
+					true, pl_obj_get_shard, layout);
+				if (leader < 0) {
+					D_WARN("Not sure whether current shard "
+					       "is leader or not for obj "DF_OID
+					       ", fseq:%d, status:%d, ver:%d, "
+					       "shard:%d, rc = %d\n",
+					       DP_OID(md->omd_id),
+					       f_shard->rfs_fseq,
+					       f_shard->rfs_status, rebuild_ver,
+					       l_shard->po_shard, leader);
+					goto fill;
+				}
+
+				rc = pool_map_find_target(map->pl_poolmap,
+							  leader, &target);
+				D_ASSERT(rc == 1);
+
+				if (myrank != target->ta_comp.co_rank) {
```

Comment:
so after this change only the leader replica will send REBUILD_OBJ req to spare target, will it cause some some obj be ignored?
for example one obj's replica 0, 1, 2 on rank A, B, C. 
On map ver 3 rank A failed, so assume will select  rank D as spare. And based on ver 3's layout will select leader rank B to send REBUILD OBJ req to spare rank D, but at that moment if rank B failed again (at ver 4) so no one will send the REBUILD OBJ req for ver 3's rebuild?
@wangdi1  is it possible?

Reply:
in this case, rebuild version 3 will not succeed, so rank b will not be marked as DOWNOUT, so during rebuild version 4, it will rebuild rank a and b at the same time.

Reply:
that obj's rebuild can be delayed to following ver 4's rebuild, probably not a problem.